docs/developer-docs/setup/deploy.mdx

```diff
@@ -0,0 +1,189 @@
+
+- #### Step 1:  Open a new terminal and navigate to your project directory.
+
+- #### Step 2:  Start the local canister execution environment, if necessary.
+
+```
+dfx start
+```
+
+- #### Step 3:  Verify the canisters you'd like to deploy are configured in the project's `dfx.json` file.
+
+An example `dfx.json` file can be found in the [default project template document](/docs/current/developer-docs/setup/first-canister)
+
+- #### Step 4:  Deploy all of the canisters within your project.
+
+To deploy locally, use the command:
+
+```
+dfx deploy 
+```
+
+To deploy to the [Motoko playground](/docs/current/developer-docs/setup/playground), use the command:
+
+```
+dfx deploy --playground
+```
+
+To deploy to the mainnet, use the command:
+
+```
+dfx deploy --network ic
+```
+
+Deploying canisters to the mainnet will cost cycles. [Learn more about cycles and how to acquire them](/docs/current/developer-docs/setup/cycles/cycles-faucet).
+
+This command deploys all canisters configured in your `dfx.json` file. To deploy just one canister, specify the canister's name:
+
+```
+dfx deploy hello_backend ## Deploy locally
+dfx deploy hello_backend --playground ## Deploy to the playground
+dfx deploy hello_backend --network ic ## Deploy to the mainnet
+```
+
+These commands both deploy to the local network since they do not include a `--network` flag. 
+
+For more information on deploying to the local or mainnet networks:
+
+- [Local deployment](/docs/current/developer-docs/setup/deploy-locally).
+
+- [Motoko playground](/docs/current/developer-docs/setup/playground).
+
+- [Mainnet deployment](/docs/current/developer-docs/setup/deploy-mainnet).
+
+## Use a custom Motoko version with `dfx deploy`
+
+To use a custom Motoko version with `dfx deploy`, export the following environment variable that indicates which Motoko base version you'd like `dfx` to use:
+
+```
+DFX_MOC_PATH="$(vessel bin)/moc" dfx deploy
+```
+
+## Setting a canister's init arguments
+
+You can set a canister's init arguments when the canister is deployed by passing the `--argument` flag in either the `dfx install` or `dfx deploy` commands:
+
+```
+dfx canister install <CANISTER_NAME> --argument "(arg in candid)"
+```
+
+```
+dfx deploy <CANISTER_NAME> --argument "(arg in candid)"
+```
+
+If several arguments should be used, an argument file can be defined with the `--argument-file` flag instead:
+
+```
+dfx deploy <CANISTER_NAME> --argument-file file.txt
+```
+
+Alternatively, init arguments can be set in `dfx.json` in `dfx` versions `v0.17.0` and newer:
+
+```json
+"canisters": {
+    "hello_backend": {
+      "candid": "src/hello_backend/hello_backend.did",
+      "package": "hello_backend",
+      "type": "rust",
+      "init_arg": "(arg in candid)"
+    },
```

Comment:
```suggestion
    },
 }
```

Reply:
FYI - I think we see this red highlighting on https://github.com/dfinity/portal/pull/2496/files because the JSON object is invalid